rust/worker/src/server.rs

```diff
@@ -189,8 +191,12 @@ impl WorkerServer {
 
         let count_orchestrator = CountQueryOrchestrator::new(
             self.clone_system()?,
-            &Uuid::parse_str(&scan.metadata_id)
-                .map_err(|e| Status::invalid_argument(e.to_string()))?,
+            &Segment::try_from(
```

Comment:
why this change?

Reply:
The Count Orchestrator is using the same query plan interface now so we have to migrate it as well. The implementation has not changed so we need to extract the information from the query plan